File: nomad-samples/js/contents-api/contents/delete-content.js
import * as prjConstants from "../constants/project-constants.js";
import apiExceptionHandler from "../exceptions/api-exception-handler.js";

export default async function deleteContent(AUTH_TOKEN, ID, CONTENT_DEFINITION_ID)
{
    const HEADERS = new Headers();
    HEADERS.append("Content-Type", "application/json");
    HEADERS.append("Authorization", `Bearer ${AUTH_TOKEN}`);

    // Send POST request
    const RESPONSE = await fetch(`${prjConstants.ADMIN_API_URL}/content/${ID}?contentDefinitionId=${CONTENT_DEFINITION_ID}`, {
        method: "DELETE",
        headers: HEADERS
    }).catch((exception) => {
        throw exception;
    });

    // Check for success
    if (RESPONSE.ok) {
        return true;
    } 

    apiExceptionHandler(RESPONSE, "Deleting movie failed");
}

File: nomad-samples/admin/py/content_api/contents/delete_content.py
from constants.project_constants import *
from exceptions.api_exception_handler import *

import requests

def delete_content(AUTH_TOKEN, ID, CONTENT_DEFINITION_ID):
    if not AUTH_TOKEN:
        raise Exception("Authorization token not found")
  
    API_URL = f"{ADMIN_API_URL}/content/{ID}?contentDefinitionId={CONTENT_DEFINITION_ID}"
    
    HEADERS = {
      "Content-Type": "application/json",
      "Authorization": "Bearer " + AUTH_TOKEN
    }

    try:
        RESPONSE = requests.delete(API_URL, headers=HEADERS)

        if (RESPONSE.ok):
            return True
        
        raise Exception()

    except:
        api_exception_handler(RESPONSE, "Deleting content failed")